### DISCUSSION TITLE
L-BFGS-B optimizer reaches to NaN errors with PyTorch Backend

### DISCUSSION THREAD

**Discussion by ehsanngh** [lululxvi/deepxde#1073] [Q&A]
Hello,

I have faced an issue when I'm using PyTorch Backend. When I switch my optimizer from Adam to "L-BFGS-B," after a few thousands iterations, I get NaN for my losses. Then, I cannot even use the best-achieved trained model for the prediction. `model.predict` returns nothing when this happens. It's worth mentioning that I don't face this issue when I have TF2 as my Backend. I was wondering maybe someone else had faced this problem before. Thanks.

![image](https://user-images.githubusercontent.com/59973930/206328126-76b9115d-a715-451b-9770-10fbc85bd01d.png)

**Comment by lululxvi** [comment]
If TF2 works, but not pytorch, then it seems an issue of the L-BFGS implementation in pytorch.
